plugin/azure/setup.go

```diff
@@ -0,0 +1,142 @@
+package azure
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	pkgFall "github.com/coredns/coredns/plugin/pkg/fall"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin/pkg/upstream"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/dns/mgmt/dns"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/caddyserver/caddy"
+)
+
+var log = clog.NewWithPlugin("azure")
+
+func init() {
+	caddy.RegisterPlugin("azure", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+
+func setup(c *caddy.Controller) error {
```

Comment:
I don't see you're restarting the go-routines to update the zones. Is this handled correctly after a reload?

Reply:
Hmm, I tried this with `reload` plugin and the updates seem to work as is. 
CMIIW but doesn't [`instance.Restart(corefile)`](https://github.com/coredns/coredns/blob/f8bba51f844a9ba9add993dcd579288d3af3707e/plugin/reload/reload.go#L83) take care of this? I don't see `routet53` needing to handle this either.

Reply:
ok